core/arch/interp.c

```diff
@@ -3506,15 +3508,17 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             }
             if (!d_r_is_avx512_code_in_use()) {
                 if (ZMM_ENABLED()) {
-                    if (instr_get_prefix_flag(bb->instr, PREFIX_EVEX)) {
-                        /* For AVX-512 detection in bb builder, we're checking only for
-                         * the prefix flag, which for example can be set by decode_cti. In
-                         * client_process_bb, post-client instructions are checked with
-                         * instr_may_write_zmm_register.
-                         */
-                        LOG(THREAD, LOG_INTERP, 2, "Detected AVX-512 code in use\n");
-                        d_r_set_avx512_code_in_use(true);
-                        proc_set_num_simd_saved(MCXT_NUM_SIMD_SLOTS);
+                    if (!instr_is_meta(bb->instr)) {
```

Comment:
That wasn't clear to me. I put in an assertion instead, also for readability. If you don't want that, please re-open.

Reply:
I would remove the assert: I think it will add confusion, because it will make it seem like all the other code looking at instructions in this app decode loop is *not* assuming these are only app instructions.

Reply:
Or put it in the general loop and not inside just this avx code.

Reply:
Will do.

Reply:
I ended up removing it. Done.